java-streams-2/src/main/java/com/baeldung/breakforeach/CustomSpliterator.java

```diff
@@ -0,0 +1,31 @@
+package com.baeldung.breakforeach;
+
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+public class CustomSpliterator extends Spliterators.AbstractSpliterator<String> {
+
+    Spliterator<String> splitr;
+    Predicate<String> predicate;
+    boolean isMatched = true;
+
+    CustomSpliterator(Spliterator<String> splitr, Predicate<String> predicate) {
+        super(splitr.estimateSize(), 0);
+        this.splitr = splitr;
+        this.predicate = predicate;
+    }
+
+    @Override
+    public boolean tryAdvance(Consumer<? super String> consumer) {
```

Comment:
Minor question - is this meant to be threadsafe? Streams can be run with `parallel`. Worth checking if this might be called by multiple threads. If that were the case, then it could be marked as `synchronized"... maybe?

Reply:
`Consumer<? extends T>`